### DISCUSSION TITLE
MongoDB-Prisma NextJS ISR (TypeScript)

### DISCUSSION THREAD

**Discussion by fanatic007** [realworld-apps/realworld#993] [WIP Implementations]
Hello folks,

I have started a new e2e implementation that I wish to announce. 
Details are as follows

**_Tech Stack_**
- Database : **MONGODB**
- ORM :  **PRISMA**
- BACKEND/API : **NEXTJS** (with **ISR** startegy)
- Frontend : **REACT**

**_Motivation_**
- I was going through NextJS ISR strategy, I want to try that with Realworl
- Prisma is having slightly different syntax for MongoDB (perhaps not implemented for Conduit)

**_Inital Progress_**
Here is the [repo](https://github.com/fanatic007/realworld_mongodb-prisma-isr) that I would love to a review on (schema and types..)
-DB Modelling; done 
-CRD APIs with Types; 

@anishkny @geromegrignon @gmlewis @StephenFluin  Hoping to get listed...

**Comment by geromegrignon** [comment]
Hi @fanatic007, thanks for your contribution to the project.

Please don't tag people randomly; some were just 'in the moment' contributors and will receive a notification about something out of their scope.

As documented [here](https://realworld-docs.netlify.app/docs/implementation-creation/expectations), we need a README file filled up with instructions about how to run ihe project.

I don't use NextJS and MongoDB, but so far I have noticed:
- token being saved in database is not required
- the :api/articles endpoint should be included in the `unless` array of unprotected endpoints.

**Reply by fanatic007** [reply]
Thanks for the quick turn around. Apologies for the tags😬. 
So with unprotected articles endpoint we just omit properties like _author.following_ and _favorited_  , right?

**Reply by geromegrignon** [reply]
Yes, that's the global feed tab content on home page which doesn't need to be logged in to be displayed.

**Comment by fanatic007** [comment]
@geromegrignon Hi ! Are the routes allowed to have query params in the url for state maintenance purpose? For eg. Home Page can be `/#/?tag=inspiration`,  `/#/feed`
I need to confirm as the demo app doesn't show any params, and was not clear to me in the guildlines
```
Home page (URL: /#/ )
List of tags
List of articles pulled from either Feed, Global, or by Tag
Pagination for list of articles
```

**Reply by geromegrignon** [reply]
Hi @fanatic007, that's not part of the specs, but I won't blame you for such an addition as planned for the next version of the specs as a valuable feature.

**Reply by fanatic007** [reply]
thanks for confirming.
